File: Python/ReverseInteger_7.py
class Solution:
    def reverse(self, x: int) -> int:
        result = -1 if x<0 else 1
        rev =int(str(abs(x))[::-1]) *result
        return rev if -(2**31) <= rev <= (2**31 - 1) else 0
    
s= Solution()

n= int(input("Enter number of test cases : "))
for i in range(n):
    a=int(input("Enter a number : "))
    result=s.reverse(a)
    print(f"Reverse Integer : {result}")

File: LeetCode/Java/ReverseInteger_7.java
import java.util.*;
class Solution7 {
    public int reverse(int x) {
        int sign = (x < 0) ? -1 : 1;
        x = Math.abs(x);
        StringBuilder s = new StringBuilder(String.valueOf(x));
        int rev = 0;
        try {
            rev = Integer.parseInt(s.reverse().toString());
        } catch (NumberFormatException e) {
            return 0;
        }
        rev *= sign;
        if (rev >= Integer.MIN_VALUE && rev <= Integer.MAX_VALUE) {
            return rev;
        }
        return 0;
    }
}

class ReverseInteger_7{
    public static void main(String[] args) {
        Scanner sc =new Scanner(System.in);
        Solution7 s=new Solution7();
        System.out.println("Enter number of test cases : ");
        int n=sc.nextInt();
        sc.nextLine();
        for(int i=0 ; i<n ;i++){
            System.out.println("Enter a number : ");
            int a=sc.nextInt();
            System.out.printf("Reversed Number : %d%n",s.reverse(a));
        }
    }
}
